### PR TITLE
chore: remove node-fetch from app code

### PR DESCRIPTION
**Pull request by mattkrick**
# Description

- Removes node-fetch since Node now has fetch natively
- Fixes the update avatar call that used `.buffer`, which doesn't exist on native fetch

## Testing scenarios

- [ ] change your user avatar

### CODE REVIEW

File: packages/client/jest.config.js [hunk 1]
```diff
@@ -3,14 +3,14 @@ const tsJestPresets = require('ts-jest/presets')
...
-  globals: {
-    'ts-jest': {
-      diagnostics: false
-    }
-  },
   transform: {
-    '\\.(gql|graphql)$': 'jest-transform-graphql',
-    ...tsJestPresets.jsWithBabel.transform
+    '\\.(gql|graphql)$': '../server/__tests__/jest-transform-graphql-shim.js',
+    '^.+\\.tsx?$': [
```
Comment (mattkrick) [inline packages/client/jest.config.js:8]: copied from https://kulshekhar.github.io/ts-jest/docs/getting-started/options/

File: packages/server/graphql/mutations/helpers/updateUserProfile.ts [hunk 2]
```diff
@@ -38,7 +37,8 @@ const updateUserProfile = async (
 
   if (validUpdatedUser.picture && validUpdatedUser.picture.endsWith('.svg')) {
     const res = await fetch(validUpdatedUser.picture)
-    const buffer = await res.buffer()
+    const arrayBuffer = await res.arrayBuffer()
```
Comment (mattkrick) [inline packages/server/graphql/mutations/helpers/updateUserProfile.ts:40]: this is the only real change

File: packages/client/package.json [hunk 3]
```diff
@@ -55,7 +55,7 @@
     "eslint-plugin-emotion": "^10.0.14",
     "eslint-plugin-react": "^7.16.0",
     "eslint-plugin-react-hooks": "^1.6.1",
-    "jest": "^27.0.6",
+    "jest": "^29.5.0",
```
Comment (mattkrick) [inline packages/client/package.json:58]: native `fetch` isn't available to jest until v28 so i bumped to latest

File: packages/client/jest.config.js (hunk 1 above)
Comment (Dschoordsch) [reply packages/client/jest.config.js:8]: +1 Should we remove the `preset` as the warning box on the page linked suggests?

**Review by Dschoordsch — APPROVED** [review]

File: packages/client/jest.config.js (hunk 1 above)
Comment (mattkrick) [reply packages/client/jest.config.js:8]: good catch! fixed

File: packages/server/graphql/mutations/helpers/updateUserProfile.ts (hunk 2 above)
Comment (mattkrick) [reply packages/server/graphql/mutations/helpers/updateUserProfile.ts:40]: right you are! we got some good spaghetti code going on there.
shall i remove that `.svg` conditional clause in this PR or should we save it for later? OK with either option.

Comment (Dschoordsch) [reply packages/server/graphql/mutations/helpers/updateUserProfile.ts:40]: Let's leave it for later and then let's remove the `picture` input from this mutation completely.

Comment (mattkrick) [reply packages/server/graphql/mutations/helpers/updateUserProfile.ts:40]: Issue made! https://github.com/ParabolInc/parabol/issues/8207

**Pull request merged**